grails-test-suite-uber/src/test/groovy/grails/validation/DomainConstraintGettersSpec.groovy

```diff
@@ -67,6 +68,7 @@ class DomainConstraintGettersSpec extends Specification implements DataTest {
 
     // DOMAIN WITH SUPER CLASS
 
+    @PendingFeature(reason = 'With Groovy 4, it is currently not possible to extend domain classes: https://issues.apache.org/jira/browse/GROOVY-5106')
```

Comment:
https://issues.apache.org/jira/browse/GROOVY-5106 seems to be fixed. are we not using a groovy version which does not contain the patch?

Reply:
@paulk-asert do you know about this?

Reply:
It is the fix to that issue that causes the problem with extending domain classes.

Reply:
If you have the following classes in Java:

    class Parent implements Iterator<CharSequence> { ... }
    class Foo extends Parent implements Iterator<String> { ... }

Java will tell you to remove one of the Iterator interface clauses:

    'java.util.Iterator' cannot be inherited with different type arguments: 'java.lang.CharSequence' and 'java.lang.String'. 

Groovy 3 allowed that to co-exist since after type erasure, it folds down onto one interface and the one you want was kept. But that was a kludge and information was lost, so Groovy 4 does the same as Java here.

In this case we have (note duplicated `GormEntity` and `Entity` with different types):

    class AbstractCustomPropertyValue implements GroovyObject, GormEntity<AbstractCustomPropertyValue>, GormEntityDirtyCheckable, WebDataBinding, DomainClass, grails.gorm.Entity<AbstractCustomPropertyValue>, DirtyCheckable.Trait.FieldHelper, GormValidateable.Trait.FieldHelper { ... }
    class StringPropertyValue extends AbstractCustomPropertyValue implements GormEntity<StringPropertyValue>, WebDataBinding, DomainClass, grails.gorm.Entity<StringPropertyValue>, DirtyCheckable.Trait.FieldHelper, GormValidateable.Trait.FieldHelper { ... }

If supporting inheritance of entities is desirable, the `@Entity` transform would need to be made smarter.  There might be a few different ways to make it work. Perhaps only abstract classes could be extended and they might not need the original interface added. Perhaps non-final classes could use a generic type like `? extends AbstractCustomProperty` (in this case).

Reply:
My example was for the previous test but the discussion still applies. For the previous test, I think you can also remove `@Entity` from the abstract parent class (and remove it from `mockDomains` too) and the test still passes prior to this PR at least.

Reply:
Yes, you are right, the test succeeds if `@Entity` is removed from `AbstractCustomPropertyValue` (and it also is removed from mockDomains).

Reply:
You could try running https://github.com/grails/grails-functional-tests project to see if anything is broken.

Reply:
> You could try running https://github.com/grails/grails-functional-tests project to see if anything is broken.

Hi @puneetbehl, is it possible to run the `grails-functional-tests` project on the this branch?

Reply:
Yes, you would need to make certain changes to the `build.gradle` file of the project.